src/main/java/org/mariadb/jdbc/MariaDbStatement.java

```diff
@@ -128,6 +128,12 @@ public MariaDbStatement(MariaDbConnection connection, int resultSetScrollType, i
     this.resultSetConcurrency = resultSetConcurrency;
     this.lock = this.connection.lock;
     this.options = this.protocol.getOptions();
+    try {
+      setFetchSize(this.options.defaultFetchSize);
+    }
+    catch(SQLException e) {
+      logger.warn("defaultFetchSize is invalid, using default of 0");
```

Comment:
having a try catch and a warning for every statement has to be avoided.

"defaultFetchSize" value must be validated beforehand, then just initialize fetchSize to defaultFetchSize. 
Best seems to be in DefaultOptions.optionCoherenceValidation or something like that.

Reply:
OK thanks, I will make the change today!

Reply:
I didn't like that either too be honest, but I didn't see that DeFaultOptions had validation. Thanks for pointing that out